tools/python_api/src_py/torch_geometric_result_converter.py

```diff
@@ -0,0 +1,237 @@
+        results = {}
+        for (i, line) in enumerate(result_str.splitlines()):
+            # ignore first line
+            if i == 0:
+                continue
+            line = line.strip()
+            if line == "":
+                continue
+            line_splited = line.split(" ")
+            if len(line_splited) < 2:
+                continue
+
+            prop_name = line_splited[0]
+            prop_type = " ".join(line_splited[1:])
+
+            is_primary_key = PRIMARY_KEY_SYMBOL in prop_type
+            prop_type = prop_type.replace(PRIMARY_KEY_SYMBOL, "")
+            depth = prop_type.count(LIST_SYMBOL)
+            prop_type = prop_type.replace(LIST_SYMBOL, "")
+            results[prop_name] = {
+                "type": prop_type,
+                "depth": depth,
+                "is_primary_key": is_primary_key
+            }
+        self.nodes_property_names_dict[table_name] = results
+        return results
+
+    def __populate_nodes_dict_and_deduplicte_edges(self):
+        self.query_result.reset_iterator()
+        while self.query_result.has_next():
+            row = self.query_result.get_next()
+            for i in self.column_to_extract:
+                column_type, _ = self.column_to_extract[i]
+                if column_type == Type.NODE.value:
+                    node = row[i]
+                    label = node["_label"]
+                    _id = node["_id"]
+                    self.table_to_label_dict[_id["table"]] = label
+
+                    if (_id["table"], _id["offset"]) in self.internal_id_to_pos_dict:
+                        continue
+
+                    node_property_names = self.__get_node_property_names(
+                        label)
+
+                    pos, primary_key = self.__extract_properties_from_node(
+                        node, label, node_property_names)
+
+                    # If no properties were extracted, then ignore the node
+                    if pos >= 0:
+                        self.internal_id_to_pos_dict[(
+                            _id["table"], _id["offset"])] = pos
+                        if label not in self.pos_to_primary_key_dict:
+                            self.pos_to_primary_key_dict[label] = {}
+                        self.pos_to_primary_key_dict[label][pos] = primary_key
+
+                elif column_type == Type.REL.value:
+                    _src = row[i]["_src"]
+                    _dst = row[i]["_dst"]
+                    self.rels.add((_src["table"], _src["offset"],
+                                   _dst["table"], _dst["offset"]))
+
+    def __extract_properties_from_node(self, node, label, node_property_names):
+        import torch
+        for prop_name in node_property_names:
+            # Ignore properties that are marked as ignored
+            if (label, prop_name) in self.ignored_columns:
+                continue
+
+            # Ignore primary key
+            if node_property_names[prop_name]["is_primary_key"]:
```

Comment:
Why not put primary_key prop_name also into the ignored properties?

Reply:
Ignored property is not read at all, but for primary key, it still needs to be read as we need to maintain the `pos_to_primary_key_dict`. I have changed the comment to `Read primary key but do not add it to the node properties`.